cmd/cli/plugin/package/main.go

```diff
@@ -23,56 +22,33 @@ var descriptor = cliv1alpha1.PluginDescriptor{
 	Group:       cliv1alpha1.RunCmdGroup,
 }
 
-var logLevel int32
-var outputFormat string
-var kubeConfig string
-
 func main() {
 	p, err := plugin.NewPlugin(&descriptor)
 	if err != nil {
 		log.Fatal(err)
 	}
 
-	p.Cmd.PersistentFlags().Int32VarP(&logLevel, "verbose", "", 0, "Number for the log level verbosity(0-9)")
-	p.Cmd.PersistentFlags().StringVarP(&kubeConfig, "kubeconfig", "", "", "The path to the kubeconfig file, optional")
-
-	p.AddCommands(
-		repositoryCmd,
-		packageInstallCmd,
-		packageAvailableCmd,
-		packageInstalledCmd,
-	)
-	if err := p.Execute(); err != nil {
+	err = nonExitingMain(p)
+	if err != nil {
 		os.Exit(1)
 	}
 }
 
-// getOutputFormat gets the desired output format for package commands that need the ListTable format
-// for its output.
-func getOutputFormat() string {
-	format := outputFormat
-	if format != string(component.JSONOutputType) && format != string(component.YAMLOutputType) {
-		// For table output, we want to force the list table format for this part
-		format = string(component.ListTableOutputType)
-	}
-	return format
-}
+func nonExitingMain(p *plugin.Plugin) error {
```

Comment:
The `main()` function uses `os.Exit()` to exit when an error is returned and therefore wouldn't run the `defer` functions. We use a function which simply returns an error instead of using `os.Exit()`, hence the name `nonExitingMain`. 
But, I see how it could be a bit confusing, let me know if you have any suggestions for an alternative :)